DataStructures.Tests/BitListTest.cs

```diff
@@ -0,0 +1,680 @@
+        {
+            var array = new BitList((sbyte)1).ToSByteArray();
+            Assert.IsTrue(array.Length == 1 && array[0] == 1);
+        }
+
+        [Test]
+        public static void ToUInt16ArrayTest()
+        {
+            var array = new BitList((ushort)1).ToUInt16Array();
+            Assert.IsTrue(array.Length == 1 && array[0] == 1);
+        }
+
+        [Test]
+        public static void ToUInt32ArrayTest()
+        {
+            var array = new BitList(1U).ToUInt32Array();
+            Assert.IsTrue(array.Length == 1 && array[0] == 1);
+        }
+
+        [Test]
+        public static void ToUInt64ArrayTest()
+        {
+            var array = new BitList(1UL).ToUInt64Array();
+            Assert.IsTrue(array.Length == 1 && array[0] == 1);
+        }
+
+        [Test]
+        public static void ToSingleArrayTest()
+        {
+            var array = new BitList(1F).ToSingleArray();
+            Assert.IsTrue(array.Length == 1 && array[0] == 1);
+        }
+
+        [Test]
+        public static void ToDoubleArrayTest()
+        {
+            var array = new BitList(1D).ToDoubleArray();
+            Assert.IsTrue(array.Length == 1 && array[0] == 1);
+        }
+
+        [Test]
+        public static void ToDecimalArrayTest()
+        {
+            var array = new BitList(1M).ToDecimalArray();
+            Assert.IsTrue(array.Length == 1 && array[0] == 1);
+        }
+
+        [Test]
+        public static void ToCharArrayTest()
+        {
+            var array = new BitList('a').ToCharArray();
+            Assert.IsTrue(array.Length == 1 && array[0] == 'a');
+        }
+
+        [Test]
+        public static void ToStringTest()
+        {
+            var array = new BitList("a").ToString();
+            Assert.IsTrue(array.Length == 1 && array == "a");
+        }
+
+        [Test]
+        public static void NotOperatorTest()
+        {
+            Assert.IsTrue(!new BitList(false) == new BitList(true));
+        }
+
+        [Test]
+        public static void BitwiseComplementOperatorTest()
+        {
+            Assert.IsTrue(~new BitList(false) == new BitList(true));
+        }
+
+        [Test]
+        public static void AndOperatorTest()
+        {
+            Assert.IsTrue((new BitList(true) & new BitList(true))[0]);
+        }
+
+        [Test]
+        public static void OrOperatorTest()
+        {
+            Assert.IsTrue((new BitList(true) | new BitList(false))[0]);
+        }
+
+        [Test]
+        public static void XorOperatorTest()
+        {
+            Assert.IsFalse((new BitList(true) ^ new BitList(true))[0]);
+        }
+
+        [Test]
+        public static void BitListCombinationOperatorTest()
+        {
+            Assert.IsTrue((new BitList(true) + new BitList(true)) == new BitList(new[] { true, true }));
+        }
+
+        [Test]
+        public static void EqualityOperatorTest()
+        {
+            Assert.IsTrue(new BitList(true) == new BitList(true));
```

Comment:
Codacy found an issue: [Correct one of the identical expressions on both sides of operator '=='.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5559704)